mk/config.mk

```diff
@@ -16,37 +16,44 @@ ifneq ($(TARGET),)
 $(error TARGET or CONFIG should be specified. Not both.)
 endif
 
-CONFIG_FILE      = $(CONFIG_DIR)/configs/$(CONFIG)/config.h
-INCLUDE_DIRS    += $(CONFIG_DIR)/configs/$(CONFIG)
+CONFIG_HEADER_FILE  = $(CONFIG_DIR)/configs/$(CONFIG)/config.h
+CONFIG_SOURCE_FILE  = $(CONFIG_DIR)/configs/$(CONFIG)/config.c
+INCLUDE_DIRS       += $(CONFIG_DIR)/configs/$(CONFIG)
 
-ifneq ($(wildcard $(CONFIG_FILE)),)
+ifneq ($(wildcard $(CONFIG_HEADER_FILE)),)
+
+CONFIG_SRC :=
+ifneq ($(wildcard $(CONFIG_SOURCE_FILE)),)
+CONFIG_SRC += $(CONFIG_SOURCE_FILE)
+TARGET_FLAGS += -DUSE_CONFIG_SOURCE
```

Comment:
Maybe do it other way around - explicitly put `USE_CONFIG_SOURCE` into config.h (fail build if config is not provided)

Reply:
I did think of that, however focused on having the defines for the functions enabled, so the linker will at least bounce if it cannot find the function (hence why I steered away from weak declarations). In this case if the file exists it will be built. If the function is required (by defining its inclusion), then there will be an error if it does not exist in the codebase.